src/Caching/StackExchangeRedis/src/RedisCache.Log.cs

```diff
@@ -0,0 +1,16 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Extensions.Caching.StackExchangeRedis;
+
+public partial class RedisCache
+{
+    private static partial class Log
+    {
+        [LoggerMessage(1, LogLevel.Warning, "Could not determine the Redis server version as the GetServer() operation is not supported.", EventName = "GetServerNotSupported")]
```

Comment:
I think the log should be less about `GetServer` failing and more about we are falling back to the deprecated `HMSET` command. Or maybe we need two logs.

Reply:
I've changed the error message to mention the different commands. I'm not tied to the specific wording, so happy to change as you deem fit.